lib/modules/platform/codecommit/index.ts

```diff
@@ -0,0 +1,718 @@
+  } catch (err) {
+    logger.error({ err }, 'findPr error');
+  }
+  if (prsFiltered.length === 0) {
+    return null;
+  }
+  return prsFiltered[0];
+}
+
+export async function getBranchPr(branchName: string): Promise<Pr | null> {
+  logger.debug(`getBranchPr(${branchName})`);
+  const existingPr = await findPr({
+    branchName,
+    state: PrState.Open,
+  });
+  return existingPr ? getPr(existingPr.number) : null;
+}
+
+export async function getPr(pullRequestId: number): Promise<Pr | null> {
+  logger.debug(`getPr(${pullRequestId})`);
+  const prRes = await client.getPr(`${pullRequestId}`);
+
+  if (!prRes?.pullRequest) {
+    return null;
+  }
+
+  const prInfo = prRes.pullRequest;
+  let prState: PrState;
+  if (prInfo.pullRequestTargets![0].mergeMetadata?.isMerged) {
+    prState = PrState.Merged;
+  } else {
+    prState =
+      prInfo.pullRequestStatus === PullRequestStatusEnum.OPEN
+        ? PrState.Open
+        : PrState.Closed;
+  }
+
+  return {
+    sourceBranch: prInfo.pullRequestTargets![0].sourceReference!,
+    state: prState,
+    number: pullRequestId,
+    title: prInfo.title!,
+    targetBranch: prInfo.pullRequestTargets![0].destinationReference!,
+    sha: prInfo.revisionId,
+  };
+}
+
+export async function getRepos(): Promise<string[]> {
+  logger.debug('Autodiscovering AWS CodeCommit repositories');
+
+  let reposRes;
+  try {
+    reposRes = await client.listRepositories();
+    //todo do we need pagination? maximum number of repos is 1000 without pagination, also the same for free account
+  } catch (error) {
+    logger.error({ error }, 'Could not retrieve repositories');
+    return [];
+  }
+
+  const res: string[] = [];
+
+  /* istanbul ignore next */
+  const repoNames = reposRes.repositories ?? [];
+
+  for (const repo of repoNames) {
+    if (repo.repositoryName) {
+      res.push(repo.repositoryName);
+    }
+  }
+
+  return res;
+}
+
+export function massageMarkdown(input: string): string {
+  // Remove any HTML we use
+  return input
+    .replace(
+      'you tick the rebase/retry checkbox',
+      'rename PR to start with "rebase!"'
+    )
+    .replace(regEx(/<\/?summary>/g), '**')
+    .replace(regEx(/<\/?details>/g), '')
+    .replace(regEx(`\n---\n\n.*?<!-- rebase-check -->.*?\n`), '')
+    .replace(regEx(/\]\(\.\.\/pull\//g), '](../../pull-requests/')
+    .replace(
+      regEx(/(?<hiddenComment><!--renovate-debug:.*?-->)/),
+      '[//]: # ($<hiddenComment>)'
+    );
+}
+
+export async function getJsonFile(
+  fileName: string,
+  repoName?: string,
+  branchOrTag?: string
+): Promise<any | null> {
+  const raw = await getRawFile(fileName, repoName, branchOrTag);
+  return raw ? JSON5.parse(raw) : null;
+}
+export async function getRawFile(
+  fileName: string,
+  repoName?: string,
+  branchOrTag?: string
+): Promise<string | null> {
+  const fileRes = await client.getFile(
+    repoName ?? config.repository,
+    fileName,
+    branchOrTag
+  );
+
+  return decoder.decode(fileRes.fileContent);
```

Comment:
```suggestion
  return fileRes.fileContent ? Buffer.from(fileRes.fileContent).toString('utf-8') : null;
```
Why about simply using `Buffer` here? So we don't need that extra dependency?

Reply:
haven't tested that, i needed an Int8array parser, ill check it out

Reply:
Buffer:
Decodes a string of Base64-encoded data into bytes, and encodes those bytes into a string using Latin-1 (ISO-8859-1).

azure has Int8Array encoded, i need something to decode it.